server/events/vcs/gitea/client.go

```diff
@@ -0,0 +1,468 @@
+func (c *GiteaClient) CreateComment(repo models.Repo, pullNum int, comment string, command string) error {
+	opt := gitea.CreateIssueCommentOption{
+		Body: comment,
+	}
+
+	_, _, err := c.giteaClient.CreateIssueComment(repo.Owner, repo.Name, int64(pullNum), opt)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ReactToComment adds a reaction to a comment.
+func (c *GiteaClient) ReactToComment(repo models.Repo, pullNum int, commentID int64, reaction string) error {
+	_, _, err := c.giteaClient.PostIssueCommentReaction(repo.Owner, repo.Name, commentID, reaction)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// HidePrevCommandComments hides the previous command comments from the pull
+// request.
+func (c *GiteaClient) HidePrevCommandComments(repo models.Repo, pullNum int, command string, dir string) error {
+	var allComments []*gitea.Comment
+
+	nextPage := int(1)
+	for {
+		// Initialize ListIssueCommentOptions with the current page
+		opts := gitea.ListIssueCommentOptions{
+			ListOptions: gitea.ListOptions{
+				Page:     nextPage,
+				PageSize: 100,
+			},
+		}
+
+		comments, resp, err := c.giteaClient.ListIssueComments(repo.Owner, repo.Name, int64(pullNum), opts)
+		if err != nil {
+			return err
+		}
+
+		allComments = append(allComments, comments...)
+
+		// Break the loop if there are no more pages to fetch
+		if resp.NextPage == 0 {
+			break
+		}
+		nextPage = resp.NextPage
+	}
+
+	currentUser, _, err := c.giteaClient.GetMyUserInfo()
+	if err != nil {
+		return err
+	}
+
+	summaryHeader := fmt.Sprintf("<!--- +-Superseded Command-+ ---><details><summary>Superseded Atlantis %s</summary>", command)
+	summaryFooter := "</details>"
+	lineFeed := "\n"
+
+	for _, comment := range allComments {
+		if comment.Poster == nil || comment.Poster.UserName != currentUser.UserName {
+			continue
+		}
+
+		body := strings.Split(comment.Body, "\n")
+		if len(body) == 0 || (!strings.Contains(strings.ToLower(body[0]), strings.ToLower(command)) && dir != "" && !strings.Contains(strings.ToLower(body[0]), strings.ToLower(dir))) {
+			continue
+		}
+
+		supersededComment := summaryHeader + lineFeed + comment.Body + lineFeed + summaryFooter + lineFeed
+
+		_, _, err := c.giteaClient.EditIssueComment(repo.Owner, repo.Name, comment.ID, gitea.EditIssueCommentOption{
+			Body: supersededComment,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// PullIsApproved returns ApprovalStatus with IsApproved set to true if the pull request has a review that approved the PR.
+func (c *GiteaClient) PullIsApproved(repo models.Repo, pull models.PullRequest) (models.ApprovalStatus, error) {
+	page := 0
+	nextPage := 1
+
+	approvalStatus := models.ApprovalStatus{
+		IsApproved: false,
+	}
+
+	listOptions := gitea.ListPullReviewsOptions{
+		ListOptions: gitea.ListOptions{
+			Page:     1,
+			PageSize: 100,
```

Comment:
hard coded, please see previous comments

Reply:
Added --gitea-page-size, defaulting to 30 based on Gitea docs.